src/main/java/com/epam/task6/data/specification/AuthorSpecification.java

```diff
@@ -0,0 +1,23 @@
+package com.epam.task6.data.specification;
+
+import com.epam.task6.model.Book;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AuthorSpecification implements Specification<String> {
+    @Override
+    public List<Book> find(List<Book> books,  String value) {
+        List<Book> searchedBooks = new ArrayList<Book>();
+
+        for(Book book: books){
```

Comment:
better make Abstact class with common logic (loop), and in child class only define method
matchFiled(Book book, T value) or
T extractFiled(Book book);
and call the method in abstarct class. see template method pattern